Update ezreader.py

rosersanchez opened the pull request:
I have:
1. Corrected startdate issue (it was computed two times, one in __init__ and the other in get_l0_data). Now we have two different attributes called "eegstartdate" and "creationdate" (referring to the capsule).
2. Called the listAttributes() function in __str__ and __repr__ and make it a little bit more readable. 

Still to correct:
1. how to access to uttilities. "ValueError: Attempted relative import in non-package".

giulioruffini commented:
Hi Roser ... not sure how this message will reach you but ... this could be
relevant:

https://stackoverflow.com/questions/11536764/how-to-fix-attempted-relative-import-in-non-package-even-with-init-py

I am not having trouble importing utilities... not so far. Hmmm

On Mon, Feb 12, 2018 at 1:43 PM, Roser Sanchez <notifications@github.com>
wrote:

> I have:
>
>    1. Corrected startdate issue (it was computed two times, one in *init*
>    and the other in get_l0_data). Now we have two different attributes called
>    "eegstartdate" and "creationdate" (referring to the capsule).
>    2. Called the listAttributes() function in *str* and *repr* and make
>    it a little bit more readable.
>
> Still to correct:
>
>    1. how to access to uttilities. "ValueError: Attempted relative import
>    in non-package".
>
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/giulioruffini/NEpy/pull/2
> Commit Summary
>
>    - Update ezreader.py
>
> File Changes
>
>    - *M* nepy/ezreader/ezreader.py
>    <https://github.com/giulioruffini/NEpy/pull/2/files#diff-0> (53)
>
> Patch Links:
>
>    - https://github.com/giulioruffini/NEpy/pull/2.patch
>    - https://github.com/giulioruffini/NEpy/pull/2.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/giulioruffini/NEpy/pull/2>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKtSdiRiP80LdXvspBDvwmX6tHYswriUks5tUIZOgaJpZM4SCo3y>
> .
>



-- 
Giulio Ruffini
President, Neuroelectrics <http://neuroelectrics.com>
__________________________________________________
Cambridge Offices:
210 Broadway, suite 201
Cambridge, MA 02139, USA
Tel +1 617 899 69 11

Barcelona Offices:
Av. Tibidabo 47bis,
08035,  Barcelona, Spain
Tel +34 93 2540362/66

mob. +34 679 49 57 77 / +1 617 899 69 11
fax.   +34 93 212 64 45 / Skype: giulio.ruffini
e-mail. giulio.ruffini@neuroelectrics.com

rosersanchez commented:
Morning, Giulio!

I'd already checked all stackoverflow pages, haha, but today I read another
reply that was really useful, thanks!

And I've solve it, I need to add this line before importing utilities:
>>sys.path.append(os.path.abspath(".."))
>>from utilities.utilities import inspectdir,
checkSignal,butter_bandpass_filter

And then when I call inspectdir i have to go back to folder "nepy" this way:
>>inspectdir("..",extension=".easy")

But I won't change it in GitHub since for you is working perfectly!

I'm starting great my day today ;)

Have a nice day you too!



Roser.


*Roser Sanchez Todo*Biophysical Modelling Specialist
*Neuroelectrics Barcelona*
__________________________________
*Barcelona Offices:*
Av. Tibidabo 47
<https://maps.google.com/?q=Av.+Tibidabo+47&entry=gmail&source=g> Bis
08035 Barcelona, Spain
Tel: +34 93 254 03 66
Fax: +34 93 212 64 45
*e-mail:* roser.sanchez@neuroelectrics.com
*website:* http://www.neuroelectrics.com

On Mon, Feb 12, 2018 at 9:24 PM, giulioruffini <notifications@github.com>
wrote:

> Hi Roser ... not sure how this message will reach you but ... this could be
> relevant:
>
> https://stackoverflow.com/questions/11536764/how-to-fix-
> attempted-relative-import-in-non-package-even-with-init-py
>
> I am not having trouble importing utilities... not so far. Hmmm
>
> On Mon, Feb 12, 2018 at 1:43 PM, Roser Sanchez <notifications@github.com>
> wrote:
>
> > I have:
> >
> > 1. Corrected startdate issue (it was computed two times, one in *init*
> > and the other in get_l0_data). Now we have two different attributes
> called
> > "eegstartdate" and "creationdate" (referring to the capsule).
> > 2. Called the listAttributes() function in *str* and *repr* and make
> > it a little bit more readable.
> >
> > Still to correct:
> >
> > 1. how to access to uttilities. "ValueError: Attempted relative import
> > in non-package".
> >
> > ------------------------------
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/giulioruffini/NEpy/pull/2
> > Commit Summary
> >
> > - Update ezreader.py
> >
> > File Changes
> >
> > - *M* nepy/ezreader/ezreader.py
> > <https://github.com/giulioruffini/NEpy/pull/2/files#diff-0> (53)
> >
> > Patch Links:
> >
> > - https://github.com/giulioruffini/NEpy/pull/2.patch
> > - https://github.com/giulioruffini/NEpy/pull/2.diff
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/giulioruffini/NEpy/pull/2>, or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AKtSdiRiP80LdXvspBDvwmX6tHYswriUks5tUIZOgaJpZM4SCo3y>
> > .
> >
>
>
>
> --
> Giulio Ruffini
> President, Neuroelectrics <http://neuroelectrics.com>
> __________________________________________________
> Cambridge Offices:
> 210 Broadway, suite 201
> <https://maps.google.com/?q=210+Broadway,+suite+201+%0D+Cambridge,+MA+02139,+USA&entry=gmail&source=g>
> Cambridge, MA 02139, USA
> Tel +1 617 899 69 11
>
> Barcelona Offices:
> Av. Tibidabo 47bis,
> 08035, Barcelona, Spain
> Tel +34 93 2540362 <+34%20932%2054%2003%2062>/66
>
> mob. +34 679 49 57 77 <+34%20679%2049%2057%2077> / +1 617 899 69 11
> fax. +34 93 212 64 45 <+34%20932%2012%2064%2045> / Skype: giulio.ruffini
> e-mail. giulio.ruffini@neuroelectrics.com
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/giulioruffini/NEpy/pull/2#issuecomment-365050468>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AijsubUt8FHhzO_widxLmpFYFKZszNVZks5tUJ3pgaJpZM4SCo3y>
> .
>

giulioruffini commented:
Ok,

I am not sure I follow *when* you have trouble with this import.

Is it when trying to use nepy in some random folder? Trying to run the
tests? not sure I follow the nature of the problem ... I am asking because
maybe I have the same problem you do, but I haven't noticed because I
haven't done the same test!

On Tue, Feb 13, 2018 at 3:18 AM, Roser Sanchez <notifications@github.com>
wrote:

> Morning, Giulio!
>
> I'd already checked all stackoverflow pages, haha, but today I read another
> reply that was really useful, thanks!
>
> And I've solve it, I need to add this line before importing utilities:
> >>sys.path.append(os.path.abspath(".."))
> >>from utilities.utilities import inspectdir,
> checkSignal,butter_bandpass_filter
>
> And then when I call inspectdir i have to go back to folder "nepy" this
> way:
> >>inspectdir("..",extension=".easy")
>
> But I won't change it in GitHub since for you is working perfectly!
>
> I'm starting great my day today ;)
>
> Have a nice day you too!
>
>
>
> Roser.
>
>
> *Roser Sanchez Todo*Biophysical Modelling Specialist
> *Neuroelectrics Barcelona*
> __________________________________
> *Barcelona Offices:*
> Av. Tibidabo 47
> <https://maps.google.com/?q=Av.+Tibidabo+47&entry=gmail&source=g>
> <https://maps.google.com/?q=Av.+Tibidabo+47&entry=gmail&source=g> Bis
> 08035 Barcelona, Spain
> Tel: +34 93 254 03 66 <+34%20932%2054%2003%2066>
> Fax: +34 93 212 64 45 <+34%20932%2012%2064%2045>
> *e-mail:* roser.sanchez@neuroelectrics.com
> *website:* http://www.neuroelectrics.com
>
> On Mon, Feb 12, 2018 at 9:24 PM, giulioruffini <notifications@github.com>
> wrote:
>
> > Hi Roser ... not sure how this message will reach you but ... this could
> be
> > relevant:
> >
> > https://stackoverflow.com/questions/11536764/how-to-fix-
> > attempted-relative-import-in-non-package-even-with-init-py
> >
> > I am not having trouble importing utilities... not so far. Hmmm
> >
> > On Mon, Feb 12, 2018 at 1:43 PM, Roser Sanchez <notifications@github.com
> >
> > wrote:
> >
> > > I have:
> > >
> > > 1. Corrected startdate issue (it was computed two times, one in *init*
> > > and the other in get_l0_data). Now we have two different attributes
> > called
> > > "eegstartdate" and "creationdate" (referring to the capsule).
> > > 2. Called the listAttributes() function in *str* and *repr* and make
> > > it a little bit more readable.
> > >
> > > Still to correct:
> > >
> > > 1. how to access to uttilities. "ValueError: Attempted relative import
> > > in non-package".
> > >
> > > ------------------------------
> > > You can view, comment on, or merge this pull request online at:
> > >
> > > https://github.com/giulioruffini/NEpy/pull/2
> > > Commit Summary
> > >
> > > - Update ezreader.py
> > >
> > > File Changes
> > >
> > > - *M* nepy/ezreader/ezreader.py
> > > <https://github.com/giulioruffini/NEpy/pull/2/files#diff-0> (53)
> > >
> > > Patch Links:
> > >
> > > - https://github.com/giulioruffini/NEpy/pull/2.patch
> > > - https://github.com/giulioruffini/NEpy/pull/2.diff
> > >
> > > —
> > > You are receiving this because you are subscribed to this thread.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/giulioruffini/NEpy/pull/2>, or mute the thread
> > > <https://github.com/notifications/unsubscribe-auth/
> > AKtSdiRiP80LdXvspBDvwmX6tHYswriUks5tUIZOgaJpZM4SCo3y>
> > > .
> > >
> >
> >
> >
> > --
> > Giulio Ruffini
> > President, Neuroelectrics <http://neuroelectrics.com>
> > __________________________________________________
> > Cambridge Offices:
> > 210 Broadway, suite 201
> <https://maps.google.com/?q=210+Broadway,+suite+201&entry=gmail&source=g>
> > <https://maps.google.com/?q=210+Broadway,+suite+201+%0D+
> Cambridge,+MA+02139,+USA&entry=gmail&source=g>
> > Cambridge, MA 02139, USA
> > Tel +1 617 899 69 11
> >
> > Barcelona Offices:
> > Av. Tibidabo 47bis,
> > 08035, Barcelona, Spain
> > Tel +34 93 2540362 <+34%20932%2054%2003%2062>
> <+34%20932%2054%2003%2062>/66
> >
> > mob. +34 679 49 57 77 <+34%20679%2049%2057%2077>
> <+34%20679%2049%2057%2077> / +1 617 899 69 11
> > fax. +34 93 212 64 45 <+34%20932%2012%2064%2045>
> <+34%20932%2012%2064%2045> / Skype: giulio.ruffini
> > e-mail. giulio.ruffini@neuroelectrics.com
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/giulioruffini/NEpy/pull/2#issuecomment-365050468>,
> or mute
> > the thread
> > <https://github.com/notifications/unsubscribe-auth/AijsubUt8FHhzO_
> widxLmpFYFKZszNVZks5tUJ3pgaJpZM4SCo3y>
> > .
> >
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/giulioruffini/NEpy/pull/2#issuecomment-365183563>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKtSdpDCAM1yOEIBQiUNEg5844xbzHoGks5tUUVIgaJpZM4SCo3y>
> .
>



-- 
Giulio Ruffini
President, Neuroelectrics <http://neuroelectrics.com>
__________________________________________________
Cambridge Offices:
210 Broadway, suite 201
Cambridge, MA 02139, USA
Tel +1 617 899 69 11

Barcelona Offices:
Av. Tibidabo 47bis,
08035,  Barcelona, Spain
Tel +34 93 2540362/66

mob. +34 679 49 57 77 / +1 617 899 69 11
fax.   +34 93 212 64 45 / Skype: giulio.ruffini
e-mail. giulio.ruffini@neuroelectrics.com

rosersanchez commented:
Okay, let’s see if I’m good at explaining.

I downloaded from GitHub the` NEpy-master.zip` and extract it in a Dropbox folder. In the first trial, I was just running the notebook “`1-ezreader demo`” with Jupyter, changing the DATA_PATH variable to my folder.

> sys.path.append('C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master-edits')

(note: I changed the backslashes from the path, since windows uses forward slashes)

When I arrived to the cell that pops an error because the `filename` is not correct, in the line that says “`These are the easy and info files I can find below the current dir:`” nothing was printed afterwards (not even the “`ValueError`” that I got later on in Spyder). 

So I opened `ezreader.py ` in Spyder and I just run the code (I wasn’t even getting any warning, which I though it is strange). And I got the first error here, this one:

> unfile('C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master/nepy/ezreader/ezreader.py', wdir='C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master/nepy/ezreader')
> Traceback (most recent call last):
> 
>   File "<ipython-input-1-125b31d1bca4>", line 1, in <module>
>     runfile('C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master/nepy/ezreader/ezreader.py', wdir='C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master/nepy/ezreader')
> 
>   File "C:\Users\roser.sanchez\Anaconda2\lib\site-packages\spyder\utils\site\sitecustomize.py", line 710, in runfile
>     execfile(filename, namespace)
> 
>   File "C:\Users\roser.sanchez\Anaconda2\lib\site-packages\spyder\utils\site\sitecustomize.py", line 86, in execfile
>     exec(compile(scripttext, filename, 'exec'), glob, loc)
> 
>   File "C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master/nepy/ezreader/ezreader.py", line 27, in <module>
>     from ..utilities.utilities import inspectdir, checkSignal,butter_bandpass_filter
> 
> ValueError: Attempted relative import in non-package

What I did next was adding “`replace`” here to make sure that I was in the right directory:

>DATA_PATH = pkg_resources.resource_filename('nepy', 'sampledata/').replace("\\","/")

But I’m still without getting any output there. I also added

>print “I’m inside inspectdir”

In the `inspectdir()` function (in the utilities folder, `utilities.py`) to see If I was importing it correctly but I wasn’t.

At the end, and after trying a lot of variations, I found that I could access “`utilities.py`” changing paths as I explained in the last comment. 

Is this useful?

giulioruffini commented:
I see!

Yes, ezreader is not meant to be run directly.

Ok, so the problem is the first one. That is the one to fix ... for some
reason it was not seeing the files in sampledata.

The problem in your notebook may be this:


# you need to change this ... which you did, ok:
sys.path.append('/Users/giulio/Dropbox/Py/dev/NEpy')
import nepy as ne

import nepy.ezreader as ez

## But you also need to change this DATA_PATH. Can you check it is created
ok? If so, try to do a >> ! ls DATA_PATH  to see if you see the files in
the sampledata folder!
# Get sample file from package:
DATA_PATH = pkg_resources.resource_filename('nepy', 'sampledata/')





On Tue, Feb 13, 2018 at 11:23 AM, Roser Sanchez <notifications@github.com>
wrote:

> Okay, let’s see if I’m good at explaining.
>
> I downloaded from GitHub theNEpy-master.zip and extract it in a Dropbox
> folder. In the first trial, I was just running the notebook “1-ezreader
> demo” with Jupyter, changing the DATA_PATH variable to my folder.
>
> sys.path.append('C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master-edits')
>
> (note: I changed the backslashes from the path, since windows uses forward
> slashes)
>
> When I arrived to the cell that pops an error because the filename is not
> correct, in the line that says “These are the easy and info files I can
> find below the current dir:” nothing was printed afterwards (not even the
> “ValueError” that I got later on in Spyder).
>
> So I opened ezreader.py in Spyder and I just run the code (I wasn’t even
> getting any warning, which I though it is strange). And I got the first
> error here, this one:
>
> unfile('C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-
> master/nepy/ezreader/ezreader.py', wdir='C:/Users/roser.sanchez/
> Dropbox/NEpy/NEpy-master/nepy/ezreader')
> Traceback (most recent call last):
>
> File "", line 1, in
> runfile('C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-
> master/nepy/ezreader/ezreader.py', wdir='C:/Users/roser.sanchez/
> Dropbox/NEpy/NEpy-master/nepy/ezreader')
>
> File "C:\Users\roser.sanchez\Anaconda2\lib\site-packages\
> spyder\utils\site\sitecustomize.py", line 710, in runfile
> execfile(filename, namespace)
>
> File "C:\Users\roser.sanchez\Anaconda2\lib\site-packages\
> spyder\utils\site\sitecustomize.py", line 86, in execfile
> exec(compile(scripttext, filename, 'exec'), glob, loc)
>
> File "C:/Users/roser.sanchez/Dropbox/NEpy/NEpy-master/nepy/ezreader/ezreader.py",
> line 27, in
> from ..utilities.utilities import inspectdir, checkSignal,butter_bandpass_
> filter
>
> ValueError: Attempted relative import in non-package
>
> What I did next was adding “replace” here to make sure that I was in the
> right directory:
>
> DATA_PATH = pkg_resources.resource_filename('nepy',
> 'sampledata/').replace("\","/")
>
> But I’m still without getting any output there. I also added
>
> print “I’m inside inspectdir”
>
> In the inspectdir() function (in the utilities folder, utilities.py) to
> see If I was importing it correctly but I wasn’t.
>
> At the end, and after trying a lot of variations, I found that I could
> access “utilities.py” changing paths as I explained in the last comment.
>
> Is this useful?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/giulioruffini/NEpy/pull/2#issuecomment-365319530>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKtSdheZkuhOg3K3-mS7iIBxHIpTdRedks5tUbcYgaJpZM4SCo3y>
> .
>



-- 
Giulio Ruffini
President, Neuroelectrics <http://neuroelectrics.com>
__________________________________________________
Cambridge Offices:
210 Broadway, suite 201
Cambridge, MA 02139, USA
Tel +1 617 899 69 11

Barcelona Offices:
Av. Tibidabo 47bis,
08035,  Barcelona, Spain
Tel +34 93 2540362/66

mob. +34 679 49 57 77 / +1 617 899 69 11
fax.   +34 93 212 64 45 / Skype: giulio.ruffini
e-mail. giulio.ruffini@neuroelectrics.com

rosersanchez commented:
Okay, Giulio, problem solved!

I could not do `! ls DATA_PATH` (since it doesn't recognize `'ls'` as an internal or external command) but I could see the files of `sampledata`  doing simply: `os.listdir(DATA_PATH)`. So at least I was sure the variable was defined properly.

The problem was that the current working directory remained unchanged, so all what `inspectdir()` was looking at where the jupyter notebooks.

So, to sum up, I have solved the "`inspectir()` issue"  by:
-  adding `.replace('\\','/')`, Windows things..
-  adding `os.chdir(DATA_PATH)`

Both changes in theory shouldn't affect a MAC user but if the user is Windows now the directory can be changed. And the `ezreader.py` remains unchanged, since I added this two lines in the jupyter demo: `1-ezreader demo`.

I'm attaching you a quick notebook explaining it better.
[howIsolveIt.pdf](https://github.com/giulioruffini/NEpy/files/1725070/howIsolveIt.pdf)

Next I'll do my last pull request of ezreader.py, and I will upload the modified `1-ezreader demo` in the library, so you can tell me what you think!

giulioruffini commented on nepy/ezreader/ezreader.py:
```diff
@@ -162,18 +162,10 @@ def __init__(self, datapath, author="anonymous"
             
         # then read data part, easy file
         self.get_l0_data()
-        timestamp = self.l0_data[self.l0_data.columns[-2]][0] 
```
Hi, why are we removing this?

giulioruffini commented on nepy/ezreader/ezreader.py:
```diff
@@ -162,18 +162,10 @@ def __init__(self, datapath, author="anonymous"
     def __repr__(self):
         #print "Capsule object created by", self.author, " on ", self.startdate
-        form = """ 
+        form = """\033[1m
```
what is this symbol ... \033[1m? Color stuff?? Cool!

rosersanchez replied:
Yes! We can also play with bold, underline... Well, I discovered and I liked it!
It can help for better visualization ;)

giulioruffini replied:
:  )

giulioruffini commented:
Roser!

I see the problem was really with the demo notebook... Maybe you can create
a demo notebook for Windows? Anyhow, great!

On Wed, Feb 14, 2018 at 12:39 PM, Roser Sanchez <notifications@github.com>
wrote:

> Okay, Giulio, problem solved!
>
> I could not do ! ls DATA_PATH (since it doesn't recognize 'ls' as an
> internal or external command) but I could see the files of sampledata
> doing simply: os.listdir(DATA_PATH). So at least I was sure the variable
> was defined properly.
>
> The problem was that the current working directory remained unchanged, so
> all what inspectdir() was looking at where the jupyter notebooks.
>
> So, to sum up, I have solved the "inspectir() issue" by:
>
>    - adding .replace('\\','/'), Windows things..
>    - adding os.chdir(DATA_PATH)
>
> Both changes in theory shouldn't affect a MAC user but if the user is
> Windows now the directory can be changed. And the ezreader.py remains
> unchanged, since I added this two lines in the jupyter demo: 1-ezreader
> demo.
>
> I'm attaching you a quick notebook explaining it better.
> howIsolveIt.pdf
> <https://github.com/giulioruffini/NEpy/files/1725070/howIsolveIt.pdf>
>
> Next I'll do my last pull request of ezreader.py, and I will upload the
> modified 1-ezreader demo in the library, so you can tell me what you
> think!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/giulioruffini/NEpy/pull/2#issuecomment-365686229>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKtSdhP5zP4lIrHiw0OrNyA6itfDDFd-ks5tUxpjgaJpZM4SCo3y>
> .
>



-- 
Giulio Ruffini
President, Neuroelectrics <http://neuroelectrics.com>
__________________________________________________
Cambridge Offices:
210 Broadway, suite 201
Cambridge, MA 02139, USA
Tel +1 617 899 69 11

Barcelona Offices:
Av. Tibidabo 47bis,
08035,  Barcelona, Spain
Tel +34 93 2540362/66

mob. +34 679 49 57 77 / +1 617 899 69 11
fax.   +34 93 212 64 45 / Skype: giulio.ruffini
e-mail. giulio.ruffini@neuroelectrics.com

rosersanchez commented:
I have done it already!
Now I was doing some last changes on ezreader.py and I will upload it all, so you can tell me what you think. Thanks!

giulioruffini commented on nepy/ezreader/ezreader.py:
```diff
@@ -162,18 +162,10 @@ def __init__(self, datapath, author="anonymous"
             
         # then read data part, easy file
         self.get_l0_data()
-        timestamp = self.l0_data[self.l0_data.columns[-2]][0] 
```
Ok... found a problem. We need to create this field on __init__ ... e.g., the __repr__ method needs to show the stardate.  So I would keep it here, then remove in the later part the defitino of eegstartdate. By the way, didn't we change these names?

rosersanchez replied:
Yes! I  just saw the same! I removed all startdates of the code, and now I'll make the program compute this value in init, got it!

giulioruffini replied:
So ... wait, I am getting confused. The new names are creationdate (maybe capsulecrationdate?) and eegstartdate. 
The methods __repr__ and __str__ are still using the self.startdate ... which does not exists. We need to create self.capsulecreationdate at __init__ and then use that attribute elswhere ... agree?

giulioruffini replied:
Hope it makes sense...

rosersanchez replied:
Totally agree!
Yes, creationdate was for the capsule and eegstartdate for the data.
And sure better to say capsulecreationdate (capsuledate, shorter? )and eegstartdate. I'll change it now!

giulioruffini commented:
Cool!

On Wed, Feb 14, 2018 at 1:16 PM, Roser Sanchez <notifications@github.com>
wrote:

> *@rosersanchez* commented on this pull request.
> ------------------------------
>
> In nepy/ezreader/ezreader.py
> <https://github.com/giulioruffini/NEpy/pull/2#discussion_r168264068>:
>
> > @@ -162,18 +162,10 @@ def __init__(self, datapath, author="anonymous"
>
>          # then read data part, easy file
>          self.get_l0_data()
> -        timestamp = self.l0_data[self.l0_data.columns[-2]][0]
>
> Totally agree!
> Yes, creationdate was for the capsule and eegstartdate for the data.
> And sure better to say capsulecreationdate (capsuledate, shorter? )and
> eegstartdate. I'll change it now!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/giulioruffini/NEpy/pull/2#discussion_r168264068>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKtSdoMeWmnTzDOz9CiHAR1ns7sf8wx-ks5tUyMOgaJpZM4SCo3y>
> .
>



-- 
Giulio Ruffini
President, Neuroelectrics <http://neuroelectrics.com>
__________________________________________________
Cambridge Offices:
210 Broadway, suite 201
Cambridge, MA 02139, USA
Tel +1 617 899 69 11

Barcelona Offices:
Av. Tibidabo 47bis,
08035,  Barcelona, Spain
Tel +34 93 2540362/66

mob. +34 679 49 57 77 / +1 617 899 69 11
fax.   +34 93 212 64 45 / Skype: giulio.ruffini
e-mail. giulio.ruffini@neuroelectrics.com
